src/main/java/org/apache/sysml/runtime/transform/RecodeAgent.java

```diff
@@ -543,13 +543,13 @@ public void initMetaData( FrameBlock meta ) {
 	}
 	
 	/**
-	 * 
-	 * @param token
-	 * @param code
-	 * @return
+	 * Returns the Recode map entry which consists of concatenation of code, delimiter and token. 
+	 * @param token	is part of Recode map
+	 * @param code  is code for token 
+	 * @return the concatenation of code and token with delimiter in between
 	 */
 	public static String constructRecodeMapEntry(String token, Long code) {
-		return token + Lop.DATATYPE_PREFIX + code.toString();
+		return code.toString() + Lop.DATATYPE_PREFIX + token;
```

Comment:
There's no need to change the format of the strings. Just search *backwards* for `Lop.DATATYPE_PREFIX` when deserializing the recode map in `getRecodeMap()`